azurelinuxagent/ga/collect_logs.py

```diff
@@ -73,11 +73,27 @@ class CollectLogsHandler(ThreadHandlerInterface):
     """
 
     _THREAD_NAME = "CollectLogsHandler"
+    __CGROUPS_FLAG_ENV_VARIABLE = "_AZURE_GUEST_AGENT_LOG_COLLECTOR_MONITOR_CGROUPS_"
```

Comment:
Is the environment variable used to distinguish between calls from the agent and calls from the command line? If yes, a command line option on the log collector (e.g. --check-cgroup <foo>) may be more appropriate.

Reply:
I'm a bit concerned with the complexity of the `azurelinuxagent.agent.parse_args` method... it doesn't quite seem tenable to keep adding flags in that manner. I do know there is an `argparse` in python3.2+, and a [PyPI package of the same name](https://pypi.org/project/argparse/) for earlier versions.

I'm not sure if it is worth switching over to this, and I'm equally unsure if my concern for adding to this `parse_args` method is warranted. Any thoughts?

((By the way, I believe this same method (of setting an environment variable) is used to pass the daemon version.))